### PR TITLE
Add `add_run_in_executor` helper to the event loop

### PR DESCRIPTION
**Pull request by penguinolog**
Implemented only for:
* `asyncio`
* `tornado` (`asyncio` as backend since 5.0)
* `select` (call executor directly)
* `glib` (call executor directly)
* `zmq` (call executor directly)

event loops at this moment

* Rework unittests for event loop for maintainability

 Fix: #575

##### Checklist
- [X] I've ensured that similar functionality has not already been implemented
- [X] I've ensured that similar functionality has not earlier been proposed and declined
- [X] I've branched off the `master` or `python-dual-support` branch
- [X] I've merged fresh upstream into my branch recently
- [X] I've ran `tox` successfully in local environment
- [X] I've included docstrings and/or documentation and/or examples for my code (if this is a new feature)

### CODE REVIEW

**Comment by penguinolog** [comment]
@danschwarz @ihabunek see added test as example for API to use for update calls from executors and callbacks usage

File: pyproject.toml [hunk 1]
```diff
@@ -47,7 +47,7 @@ classifiers = {file = ["classifiers.txt"]}
 [project.optional-dependencies]
 curses = ["windows-curses;sys_platform=='win32'"]
 glib = ["PyGObject"]
-tornado = ["tornado"]
```
Comment (penguinolog) [inline pyproject.toml:50]: <5.0 is not supported since version 2.2.0

**Comment by penguinolog** [comment]
Thecnically it was possible to just expose `_loop` from Asyncio/Tornado event loops, but it will be not so comfortable in IDE's and static checking.
Public API is almost the same (small deviations).

**Comment by wardi** [comment]
Maybe an update for the docs/examples to indicate when and how to use run_in_executor?

**Comment by wardi** [comment]
So how does this change a program that would have called the MainLoop.run method?

**Comment by penguinolog** [comment]
> So how does this change a program that would have called the MainLoop.run method?

Change is needed only if threading is used: by design `asyncio` and `tornado` should call executors using `run_in_executor`. Normally it's associated with event loop blocking, but in our case it associated with not triggering callback set from other thread if nothing happens in main thread (no input and no scheduled event).
As I commented bug, instead of calling `executor.submit` (where executor is `concurrent.futures.ThreadPoolExecutor` or `concurrent.futures.ProcessPoolExecutor`) need to call `evl.run_in_executor` where `evl` is our EventLoop.
Also specially explained `TwistedEventLoop` case (it implements it's own thread pool and will crash in case of usage external executors)

**Comment by wardi** [comment]
Do you think it's worth adding a small example program that demonstrates e.g. using urwid with a worker thread, asyncio and run_in_executor?

**Comment by penguinolog** [comment]
> Do you think it's worth adding a small example program that demonstrates e.g. using urwid with a worker thread, asyncio and run_in_executor?

We can add fixed program from bug, but it's a special case IMHO. We can go too deep in examples.

**Review by wardi — APPROVED** [review]

**Pull request merged**